### ISSUE TITLE
CollectDuplexSeqMetrics error

### ISSUE THREAD

**Issue by xujishu** [fulcrumgenomics/fgbio#433]
I got error when I ran `CollectDuplexSeqMetrics` with the bam files output from`GroupReadsByUmi` and got the following errors.  The header of bam files show bam files are sorted as  `@HD	VN:1.5	SO:unsorted	GO:query	SS:unsorted:template-coordinate`.  Anybody can tell me how to fix this error?

Thanks 

[2018/09/18 08:56:17 | FgBioMain | Info] Executing CollectDuplexSeqMetrics from fgbio version 0.7.0-f2d5d5e-SNAPSHOT as jishu@jishu-XPS-15-9560 on JRE 10.0.2+13-Ubuntu-1ubuntu0.18.04.2 with snappy
[2018/09/18 08:56:17 | FgBioMain | Info] CollectDuplexSeqMetrics failed. Elapsed time: 0.03 minutes.
Exception in thread "main" scala.MatchError: [Ljava.lang.String;@788ddc1f (of class [Ljava.lang.String;)
	at com.fulcrumgenomics.umi.CollectDuplexSeqMetrics.$anonfun$updateUmiMetrics$3(CollectDuplexSeqMetrics.scala:392)
	at scala.collection.Iterator.foreach(Iterator.scala:929)
	at scala.collection.Iterator.foreach$(Iterator.scala:929)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
	at com.fulcrumgenomics.umi.CollectDuplexSeqMetrics.updateUmiMetrics(CollectDuplexSeqMetrics.scala:392)
	at com.fulcrumgenomics.umi.CollectDuplexSeqMetrics.$anonfun$collect$6(CollectDuplexSeqMetrics.scala:364)
	at com.fulcrumgenomics.umi.CollectDuplexSeqMetrics.$anonfun$collect$6$adapted(CollectDuplexSeqMetrics.scala:355)
	at scala.collection.immutable.Stream.foreach(Stream.scala:530)
	at com.fulcrumgenomics.umi.CollectDuplexSeqMetrics.$anonfun$collect$2(CollectDuplexSeqMetrics.scala:355)
	at scala.runtime.java8.JFunction1$mcVD$sp.apply(JFunction1$mcVD$sp.java:12)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:32)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:29)
	at scala.collection.mutable.ArrayOps$ofDouble.foreach(ArrayOps.scala:275)
	at com.fulcrumgenomics.umi.CollectDuplexSeqMetrics.collect(CollectDuplexSeqMetrics.scala:349)
	at com.fulcrumgenomics.umi.CollectDuplexSeqMetrics.execute(CollectDuplexSeqMetrics.scala:326)
	at com.fulcrumgenomics.cmdline.FgBioMain.makeItSo(FgBioMain.scala:99)
	at com.fulcrumgenomics.cmdline.FgBioMain.makeItSoAndExit(FgBioMain.scala:80)
	at com.fulcrumgenomics.cmdline.FgBioMain$.main(FgBioMain.scala:48)
	at com.fulcrumgenomics.cmdline.FgBioMain.main(FgBioMain.scala)

**Comment by nh13** [comment]
@xujishu it looks like it is failing to parse the value in the `RX` tag for a read.  It is expecting two umi sequences separated (delimited) by a dash. (ex `AAACCC-CCCAAA`).  Can you verify the format of the `RX` tags?  Can you post a few reads for us to inspect?

**Comment by xujishu** [comment]
Here are some reads. 
The bam file was aligned by HISAT2 first then UMI tag was added by `AnnotateBamWithUmis` 


A00477:12:H3YM7DSXX:2:2307:30318:2863	99	chr1	11800	60	46M170380N80M	=	182226	170544	CTTTGCTGTTCCTGCATGTAGTTTAAACGAGATTGCCAGCACCGGGTGGGGCCTGGCCATGTGTATTTTTTTAAATTTCCACTGATGATTTTGCTGCATGGCCGGTGTTGAGAATGACTGCGCAAA	FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MC:Z:9S80M1D37M	MD:Z:126	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:0	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:-16	XS:A:-	YS:i:-17	YT:Z:CP	RX:Z:GCGCAACTT
A00477:12:H3YM7DSXX:2:2307:30318:2863	147	chr1	182226	60	9S80M1D37M	=	11800	-170544	AGCACCGGGTGGGGCCTGGCCATGTGTATTTTTTTAAATTTCCACTGATGATTTTGCTGCATGGCCGGTGTTGAGAATGACTGCGCAAATTGCCGGATTTCCTTTGCTGTTCCTGCATGTAGTTTA	FFFFFFFFFFF:FFFFF:F,FFFF:FFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MC:Z:46M170380N80M	MD:Z:80^T37	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:1	NH:i:1	MI:Z:0	NM:i:1	XM:i:0	XN:i:0	XO:i:1	MQ:i:60	AS:i:-17	XS:A:-	YS:i:-16	ZS:i:-17	YT:Z:CP	RX:Z:GCGCAACTT
A00477:12:H3YM7DSXX:4:2638:22788:14027	83	chr1	13063	60	126M	=	13007	-182	CTGCCTCCAGCTGGGTCGACAGACAGGGGCTGGAGAAGGGGAGAAGAGGAAAGTGAGGTTGCCTGCCCTGTCTCCTACCTGAGGCTGAGGAAGGAGAAGGGGATGCACTGTTGGGGAGGCAGCTGT	FFFFFFFFFFFFFFFFFFFF:F:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MC:Z:126M	MD:Z:126	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:1	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	ZS:i:-10	YT:Z:CP	RX:Z:CCACGCTGG
A00477:12:H3YM7DSXX:4:2638:22788:14027	163	chr1	13007	60	126M	=	13063	182	GCTCTGGTGGAGAACCTGTGCATGAAGGCTGTCAACCAGTCCATAGGCAAGCCTGGCTGCCTCCAGCTGGGTCGACAGACAGGGGCTGGAGAAGGGGAGAAGAGGAAAGTGAGGTTGCCTGCCCTG	FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFF,FFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MC:Z:126M	MD:Z:126	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:1	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	YT:Z:CP	RX:Z:CCACGCTGG
A00477:12:H3YM7DSXX:2:2465:17815:10864	83	chr1	13062	60	126M	=	13039	-149	GCTGCCTCCAGCTGGGTCGACAGACAGGGGCTGGAGAAGGGGAGAAGAGGAAAGTGAGGTTGCCTGCCCTGTCTCCTACCTGAGGCTGAGGAAGGAGAAGGGGATGCACTGTTGGGGAGGCAGCTG	FFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MC:Z:126M	MD:Z:126	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:2	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	ZS:i:-11	YT:Z:CP	RX:Z:GGGTTATGT
A00477:12:H3YM7DSXX:2:2465:17815:10864	163	chr1	13039	60	126M	=	13062	149	CAACCAGTCCATAGGCAAGCCTGGCTGCCTCCAGCTGGGTCGACAGACAGGGGCTGGAGAAGGGGAGAAGAGGAAAGTGAGGTTGCCTGCCCTGTCTCCTACCTGAGGCTGAGGAAGGAGAAGGGG	FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,FFFFFFFFFF	MC:Z:126M	MD:Z:126	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:2	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	YT:Z:CP	RX:Z:GGGTTATGT
A00477:12:H3YM7DSXX:4:2211:25988:26021	83	chr1	13337	60	81M4I40M	=	13114	-344	AAAGCCACCTTTGGTTCTGCCATTGCTGCTGTGTGGAAGTTCACTCCTGCCTTTTCCTTTCCCTAGAGCCTCCACCACCCCGAGAGAGATCACATTTCTCACTGCCTTTTGTCTGCCCAGTTTCA	FFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF::FFFFFFFFFFFFFFFFFFFFF:FFFFFFF,FFFFFFFFFF,FFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MC:Z:126M	MD:Z:121	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:3	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	YT:Z:CP	RX:Z:CGCTTTACA	Zs:Z:81|I|rs777038595
A00477:12:H3YM7DSXX:4:2211:25988:26021	163	chr1	13114	60	126M	=	13337	344	AGTGAGGTTGCCTGCCCTGTCTCCTACCTGAGGCTGAGGAAGGAGAAGGGGATGCACTGTTGGGGAGGCAGCTGTAACTCAAAGCCTTAGCCTCTGTTCCCACGAAGGCAGGGCCATCAGGCACCA	:FFFFFFFFFFFFFFFF:FFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFF:FFFFF,FFFFFFFFFFFFFFFF:F,	MC:Z:81M4I40M	MD:Z:126	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:3	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	ZS:i:0	YT:Z:CP	RX:Z:CGCTTTACA
A00477:12:H3YM7DSXX:2:1613:9887:7373	83	chr1	13229	60	126M	=	13161	-194	ATCAGGCACCAAAGGGATTCTGCCAGCATAGTGCTCCTGGACCACTGATACACCCGGCACCCTGTCCTGGACACGCTGTTGGCCTGGATCTGAGCCCTGGTGGAGGTCAAAGCCACCTTTGGTTCT	FFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFF	MC:Z:126M	MD:Z:44G81	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:4	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	YT:Z:CP	RX:Z:TAGTAAGTG	Zs:Z:44|S|rs531730856
A00477:12:H3YM7DSXX:2:1613:9887:7373	163	chr1	13161	60	126M	=	13229	194	GGGGATGCACTGTTGGGGAGGCAGCTGTAACTCAAAGCCTTAGCCTCTGTTCCCACGAAGGCAGGGCCATCAGGCACCAAAGGGATTCTGCCAGCATAGTGCTCCTGGACCACTGATACACCCGGC	FFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFF:FFFF:FFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFF,FFFFFFFFFFFFFFFFFF	MC:Z:126M	MD:Z:112G13	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:4	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	ZS:i:-14	YT:Z:CP	RX:Z:TAGTAAGTG	Zs:Z:112|S|rs531730856
A00477:12:H3YM7DSXX:1:1469:21070:28401	83	chr1	13236	60	126M	=	13174	-188	ACCAAAGGGATTCTGCCAGCATAGTGCTCCTGGACCAGTGATACACCCGGCACCCTGTCCTGGACACGCTGTTGGCCTGGATCTGAGCCCTGGTGGAGGTCAAAGCCACCTTTGGTTCTGCCATTG	FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MC:Z:126M	MD:Z:126	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:5	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	YT:Z:CP	RX:Z:AATCAATCG
A00477:12:H3YM7DSXX:1:1469:21070:28401	163	chr1	13174	60	126M	=	13236	188	TGGGGAGGCAGCTGTAACTCAAAGCCTTAGCCTCTGTTCCCACGAAGGCAGGGCCATCAGGCACCAAAGGGATTCTGCCAGCATAGTGCTCCTGGACCAGTGATACACCCGGCACCCTGTCCTGGA	FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFF:FF:FFFFFFFFFFFF:FF	MC:Z:126M	MD:Z:126	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:5	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	ZS:i:0	YT:Z:CP	RX:Z:AATCAATCG
A00477:12:H3YM7DSXX:4:1371:28953:34632	83	chr1	13274	60	126M	=	13187	-213	TGATACACCCGGCACCCTGTCCTGGACACGCTGTTGGCCTGGATCTGAGCCCTGGTGGAGGTCAAAGCCACCTTTGGTTCTGCCATTGCTGCTGTGTGGAAGTTCACTCCTGCCTTTTCCTTTCCC	FFF,FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFF,FFFF,FFFF:F,FFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFF,FFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MC:Z:124M	MD:Z:126	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:6	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	YT:Z:CP	RX:Z:GAATGTAGT
A00477:12:H3YM7DSXX:4:1371:28953:34632	163	chr1	13187	60	124M	=	13274	213	GTAACTCAAAGCCTTAGCCTCTGTTCCCACGAAGGCAGGGCCATCAGGCACCAAAGGGATTCTGCCAGCATAGTGCTCCTGGACCACTGATACACCCGGCACCCTGTCCTGGACACGCTGTTGG	FFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFF,FFFFFFFFFFFFFFFFFF:FFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,:FFFFFFFFFFFFF:F,FFFFF:FF:FF	MC:Z:126M	MD:Z:86G37	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:6	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	YT:Z:CP	RX:Z:GAATGTAGT	Zs:Z:86|S|rs531730856
A00477:12:H3YM7DSXX:1:2212:14335:10003	83	chr1	13234	60	112M	=	13234	-112	GCACCAAAGGGATTCTGCCAGCATAGTGCTCCTGGACCACTGATACACCCGGCACCCTGTCCTGGACACGCTGTTGGCCTGGATCTGAGCCCTGGTGGAGGTCAAAGCCACC	FFFFFFF:FFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MC:Z:112M	MD:Z:39G72	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:7	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	YT:Z:CP	RX:Z:TCTACATTT	Zs:Z:39|S|rs531730856
A00477:12:H3YM7DSXX:1:2212:14335:10003	163	chr1	13234	60	112M	=	13234	112	GCACCAAAGGGATTCTGCCAGCATAGTGCTCCTGGACCACTGATACACCCGGCACCCTGTCCTGGACACGCTGTTGGCCTGGATCTGAGCCCTGGTGGAGGTCAAAGCCACC	FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MC:Z:112M	MD:Z:39G72	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:7	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	YT:Z:CP	RX:Z:TCTACATTT	Zs:Z:39|S|rs531730856
A00477:12:H3YM7DSXX:2:2128:13630:11600	83	chr1	13296	60	126M	=	13264	-158	TGGACACGCTGTTGGCCTGGATCTGAGCCCTGGTGGAGGTCAAAGCCACCTTTGGTTCTGCCATTGCTGCTGTGTGGAAGTTCACTCCTGCCTTTTCCTTTCCCTAGAGCCTCCACCACCCCGAGA	FFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFF	MC:Z:126M	MD:Z:126	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:8	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	ZS:i:-5	YT:Z:CP	RX:Z:CCCCCCCCC
A00477:12:H3YM7DSXX:2:2128:13630:11600	163	chr1	13264	60	126M	=	13296	158	CCTGGACCACTGATACACCCGGCACCCTGTCCTGGACACGCTGTTGGCCTGGATCTGAGCCCTGGTGGAGGTCAAAGCCACCTTTGGTTCTGCCATTGCTGCTGTGTGGAAGTTCACTCCTGCCTT	FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFF,FFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFF,FFFFFFFFFFF:FFFFFFFF:FFFFF,FFFFFF:FFFFFFF:	MC:Z:126M	MD:Z:9G116	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:8	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	YT:Z:CP	RX:Z:CCCCCCCCC	Zs:Z:9|S|rs531730856
A00477:12:H3YM7DSXX:1:2611:20636:21386	83	chr1	13359	60	59M4I62M	=	13356	-124	TTGCTGCTGTGTGGAAGTTCACTCCTGCCTTTTCCTTTCCCTAGAGCCTCCACCACCCCGAGAGAGATCACATTTCTCACTGCCTTTTGTCTGCCCAGTTTCACCAGAAGTAGGCCTCTTCCTGA	:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MC:Z:62M4I60M	MD:Z:121	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:9	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	YT:Z:CP	RX:Z:GCTCGAATA	Zs:Z:59|I|rs777038595
A00477:12:H3YM7DSXX:1:2611:20636:21386	163	chr1	13356	60	62M4I60M	=	13359	124	CCATTGCTGCTGTGTGGAAGTTCACTCCTGCCTTTTCCTTTCCCTAGAGCCTCCACCACCCCGAGAGAGATCACATTTCTCACTGCCTTTTGTCTGCCCAGTTTCACCAGAAGTAGGCCTCTTCCT	FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF	MC:Z:59M4I62M	MD:Z:122	RG:Z:02552539_Cer_tR_NEB300UMI	XG:i:0	NH:i:1	MI:Z:9	NM:i:0	XM:i:0	XN:i:0	XO:i:0	MQ:i:60	AS:i:0	XS:A:+	YS:i:0	YT:Z:CP	RX:Z:GCTCGAATA	Zs:Z:62|I|rs777038595

**Comment by nh13** [comment]
1. See my comment about the `RX` tag not storing two UMIs. Are your data duplex seq?  If so, they should have two UMIs per read, which are stored in the `RX` tag delimited by a dash.
2. Did you run `GroupReadsByUmi` using the paired strategy (i.e. `--strategy=Paired`)?   It should have failed at that step given you only have one UMI in each read.

**Comment by xujishu** [comment]
1) the read structure is like this: Paired-end reads, R1 and R2, only R1 is tagged with UMI, so it would like R1+UMI and R2. so it is only one UMI
2) I used `adjacency` with edit distance 1 when I ran `GroupReadsByUmi`

The ovreall workflow is like this
0) IDT dual index+UMI RNA-Seq protocol. 
1) extract  R1 R2 and UMI from basecall. then run HISAT2 to align R1 and R2. output aligned.bam
2) Use `AnnotateBamWithUmis` to add UMI tag to aligned bam, then sort by query name if necessary
3) Run SetMateInformation to set mate CIGAR tag
4) Run `GroupReadsByUmi`  

the error happened when I tried to run `CollectDuplexSeqMetrics`
